### PR TITLE
Added records instead of relying on bit twiddling

### PR DESCRIPTION
**Pull request by FredPraca**
Hello Nick,
I made a lot of changes to externalize inner packages of s-stm32f.ads and make use of records with representation clauses instead of using word type everywhere.
The reason for this is that I'm not fond of playing with raw bit fields and prefer using more readable records.

Feel free to incorporate this if you want.

Fred

### CODE REVIEW

File: ravenscar-sfp-stm32f411-nucleo/adainclude/s-stmgpi.ads [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+   for PUPDR use (No_Pull => 0,
+                  Pull_Up => 1,
+                  Pull_Down => 2);
+
+   type PUPDR_IOs is array (IO_Index) of PUPDR with Pack;
+
+   type Inputs is array (IO_Index) of Boolean with Pack;
+   type Outputs is array (IO_Index) of Boolean with Pack;
+
+   type LCKn is array (IO_Index) of Boolean with Pack;
```
Comment (nickpascucci) [inline ravenscar-sfp-stm32f411-nucleo/adainclude/s-stmgpi.ads]: Should this just be "LCK"?

**Comment by nickpascucci** [comment]
This is awesome, thanks for doing this work! I have a few comments on minor issues, but overall this looks fantastic and I look forward to merging it.

**Comment by FredPraca** [comment]
Thanks for all your comments and for the time you spent on it.
I'll try to fix things as soon as possible but I'll be far from home the next two weeks so I don't know if I'll be able to work on it. 
Moreover I still have some problems linked to the representation clauses and side-effects when changing a component of a record instead of the whole record.
Thanks again

**Comment by FredPraca** [comment]
This should be ok now.

**Comment by nickpascucci** [comment]
Thanks for the contribution!

**Pull request merged**